### ISSUE TITLE
[3.9] Array and string offset access syntax with curly braces is deprecated

### ISSUE THREAD

**Issue by sshcli** [joomla/joomla-cms#28900]
### Steps to reproduce the issue
1. Install Joomla 3.9.18 + PHP 7.4.5
2. Visit your website and spend a few seconds navigating between pages
3. Take a look to the error log file

### Expected result
No errors

### Actual result
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /home/username/public_html/libraries/vendor/leafo/lessphp/lessc.inc.php on line 749
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /home/username/public_html/libraries/vendor/leafo/lessphp/lessc.inc.php on line 1860
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /home/username/public_html/libraries/vendor/leafo/lessphp/lessc.inc.php on line 2519
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /home/username/public_html/libraries/vendor/leafo/lessphp/lessc.inc.php on line 2573

### System information (as much as possible)
PHP 7.4.5
Joomla 3.9.18

**Comment by brianteeman** [comment]
Although the file is distributed with Joomla there is nothing in the core that uses it. It is probably being used by your template. It absolutely should **not** be enabled by default. You would only want to be using it when you have changed your template - and then only once.

**Comment by sshcli** [comment]
You are right @brianteeman 

Thanks @brianteeman for point me into the right direction.

Thanks @810 for give me the solution.

I'm closing this issue now

**Comment by MrTobyDog** [comment]
I have just update to 3.9.18 and get the following (PHP 7.4.3):

Deprecated: Array and string offset access syntax with curly braces is deprecated in I:\Users\Dave\WebSites\J3918\libraries\joomla\database\driver.php on line 1956

Deprecated: Array and string offset access syntax with curly braces is deprecated in I:\Users\Dave\WebSites\J3918\libraries\joomla\database\driver.php on line 1956

Deprecated: Array and string offset access syntax with curly braces is deprecated in I:\Users\Dave\WebSites\J3918\libraries\joomla\database\driver.php on line 1956

Deprecated: Array and string offset access syntax with curly braces is deprecated in I:\Users\Dave\WebSites\J3918\libraries\joomla\database\driver.php on line 1956

Deprecated: Array and string offset access syntax with curly braces is deprecated in I:\Users\Dave\WebSites\J3918\libraries\joomla\database\driver.php on line 1956

Deprecated: Array and string offset access syntax with curly braces is deprecated in I:\Users\Dave\WebSites\J3918\libraries\joomla\database\driver.php on line 2032

Deprecated: Array and string offset access syntax with curly braces is deprecated in I:\Users\Dave\WebSites\J3918\libraries\src\Filesystem\Path.php on line 148

Deprecated: Array and string offset access syntax with curly braces is deprecated in I:\Users\Dave\WebSites\J3918\libraries\src\Filesystem\Path.php on line 151

Deprecated: Array and string offset access syntax with curly braces is deprecated in I:\Users\Dave\WebSites\J3918\libraries\src\Filesystem\Path.php on line 154<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28900">issues.joomla.org/tracker/joomla-cms/28900</a>.</sub>

**Comment by zero-24** [comment]
Hi please open a new issue with that warnings so we can take a look into them.

**Comment by zero-24** [comment]
Just looked them up on 3.9.18 @MrTobyDog seems you are not running a real 3.9.18 please reinstall the core files from com_joomlaupdate and try again and check the content to up-to-date with 3.9.18 stable.

> Deprecated: Array and string offset access syntax with curly braces is deprecated in I:\Users\Dave\WebSites\J3918\libraries\joomla\database\driver.php on line 1956

https://github.com/joomla/joomla-cms/blob/3.9.18/libraries/joomla/database/driver.php#L1956

> Deprecated: Array and string offset access syntax with curly braces is deprecated in I:\Users\Dave\WebSites\J3918\libraries\joomla\database\driver.php on line 2032

https://github.com/joomla/joomla-cms/blob/3.9.18/libraries/joomla/database/driver.php#L2032

> Deprecated: Array and string offset access syntax with curly braces is deprecated in I:\Users\Dave\WebSites\J3918\libraries\src\Filesystem\Path.php on line 148

https://github.com/joomla/joomla-cms/blob/3.9.18/libraries/src/Filesystem/Path.php#L148

 > Deprecated: Array and string offset access syntax with curly braces is deprecated in I:\Users\Dave\WebSites\J3918\libraries\src\Filesystem\Path.php on line 151

https://github.com/joomla/joomla-cms/blob/3.9.18/libraries/src/Filesystem/Path.php#L151

> Deprecated: Array and string offset access syntax with curly braces is deprecated in I:\Users\Dave\WebSites\J3918\libraries\src\Filesystem\Path.php on line 154

https://github.com/joomla/joomla-cms/blob/3.9.18/libraries/src/Filesystem/Path.php#L154

**Comment by MrTobyDog** [comment]
Thank you - I had reached the same conclusion! I originally downloaded the
3.9.18 update files from the link on the update screen on my website and
they identified as such, but after I got this error I checked re-installed
that update and noticed it was 3.9.11 although the name was 3.9.18! I
downloaded the file again from the Joomla site and it was OK - so it seems
as though the link on the update screen MAY have been wrong?

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by zero-24** [comment]
That sounds interesting. Please send me a mail with the details to tobias.zulauf[at]community.joomla.org so i can take a look and make sure this is not a general problem. Thanks.

**Comment by sabassanchez** [comment]
Not sure if there was a solution to this but I was able to search around and fix the issue by going to each line and replacing every 

if ($query != '' && $query{0} != '#')
TO
if ($query != '' && $query[0] != '#')

Basically, replace the {} to [] on all the error lines


SOURCE:
https://joomdonation.com/forum/membership-pro-general-discussion/67120-array-and-string-offset-access-syntax-with-curly-braces-is-deprecated-in-php-7-4.html<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28900">issues.joomla.org/tracker/joomla-cms/28900</a>.</sub>

**Comment by MrTobyDog** [comment]
Yes I did that. But fixed in latest release
